### PR TITLE
Update pubads_impl regex to cover edge cases.

### PR DESCRIPTION
**Pull request by jburger424**

### CODE REVIEW

File: lighthouse-plugin-publisher-ads/test/utils/resource-classification_test.js [hunk 1]
```diff
@@ -163,4 +163,34 @@ describe('resource-classification', () => {
...
+  describe('#isGptImplTag', () => {
+    const testCases = [
+      {
+        description: 'Standard tag matches',
+        url: new URL('https://securepubads.g.doubleclick.net/gpt/pubads_impl_19700101.js?1234'),
+        expectation: true,
+      },
+      {
+        description: 'Non-standard tag matches',
+        url: new URL('https://securepubads.g.doubleclick.net/gpt/pubads_impl_queso_19700101.js'),
```
Comment (warrengm) [inline lighthouse-plugin-publisher-ads/test/utils/resource-classification_test.js]: s/queso/modern/ for a real world example

Comment (jburger424) [reply lighthouse-plugin-publisher-ads/test/utils/resource-classification_test.js]: fine...
